### ISSUE TITLE
Frontend input date field reformats date incorrect

### ISSUE THREAD

**Issue by jacob-v-dam** [forkcms/forkcms#1174]
I created a custom form on the frontend with the following code:
```php
$date = new \DateTime('today');
$form->addDate('start_date', $date->getTimestamp());
```

On the frontend in the code the date is written as `02/05/2015` but jQuery changes it to `05/02/2015` which is wrong because today is `02/05/2015`. I can't find the problem, it happens since I updated from 3.8.7 to 3.9.1.

**Comment by WouterSioen** [comment]
Have you checked if your date format is set correctly in the backend (it's in the settings module).

**Comment by jacob-v-dam** [comment]
Well the date field Doesn't respect the date/time settings of Fork CMS. It converts them to `d/m/Y`.

**Comment by WouterSioen** [comment]
Looks like this is still a todo: https://github.com/forkcms/forkcms/blob/master/src/Frontend/Core/Engine/Form.php#L164

Should not be that hard to fetch it from the fork.settings service and add it as data-mask.

**Comment by WouterSioen** [comment]
@jacob-v-dam should be fixed in our master branch (and in our upcoming release)

**Comment by dubgeiser** [comment]
It is not.  The problem is in the interpretation of the mask, not where the mask comes from.
![screen shot 2015-06-04 at 11 59 34](https://cloud.githubusercontent.com/assets/290589/7981771/777ce330-0ab1-11e5-89a1-9e849fd73f8b.png)
![screen shot 2015-06-04 at 11 59 42](https://cloud.githubusercontent.com/assets/290589/7981772/778f77ac-0ab1-11e5-9c99-13c6f6a4e55d.png)

**Comment by WouterSioen** [comment]
@ikoene can you check this please? I guess the conversion on these lines: https://github.com/forkcms/forkcms/blob/b9bf2a2789b5a959d39265934f3d107444764a80/src/Frontend/Core/Engine/Form.php#L169-L173 is not correct then.

**Comment by WouterSioen** [comment]
I guess 'Y' should be converted to 'yyyy' instead of 'yy'

**Comment by ikoene** [comment]
@WouterSioen Tested that and it's not correct. the 'yyyy' format displays the year twice.

The value is actually set twice. Once when parsing the form and a second time when the date fields are initialized in frontend.js

I think the problem could be located here:
https://github.com/forkcms/forkcms/blob/master/src/Frontend/Core/Js/frontend.js#L261

The formatted value is not correct. Sometimes it switches the day with the month. Strangely I also had different results in different browsers, including @dubgeiser 's example.

**Comment by WouterSioen** [comment]
Strange, when looking at the docs for https://api.jqueryui.com/datepicker/, it lookds like the formatDate method is used correctly.

**Comment by ikoene** [comment]
var phpDate = new Date($(this).val()); // Get date from php in YYYY-MM-DD format

My guess is the phpDate value is not correctly formatted.

**Comment by ikoene** [comment]
Dumped the value

    Mon Apr 06 2015 00:00:00 GMT+0200 (CEST)

**Comment by WouterSioen** [comment]
We should maybe pass all items as paramters this way:

`new Date(2015, 6, 4, 0, 0, 0);`

**Comment by WouterSioen** [comment]
Or use another dateformat. Reference: http://dygraphs.com/date-formats.html

**Comment by ikoene** [comment]
Fixed in #1227

**Comment by WouterSioen** [comment]
Should be really fixed now :+1:
